libcaf_net/caf/net/lp/framing.cpp

```diff
@@ -186,14 +201,17 @@ class framing_impl : public framing {
   upper_layer_ptr up_;
 
   size_t message_offset_ = 0;
+
+  size_t max_message_length_ = 0;
```

Comment:
Again, this is not about a length check. This is about configuring how many bytes we are reading from / writing to the wire! This version still uses 4 bytes unconditionally for the length.